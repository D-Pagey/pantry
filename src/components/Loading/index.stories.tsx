import { Loading } from '.';

export default { title: 'Loading' };

export const normal = (): JSX.Element => <Loading isLoading />;
