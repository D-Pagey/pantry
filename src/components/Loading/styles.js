import styled from 'styled-components';
import { MOBILE_HEADER_HEIGHT, HEADER_HEIGHT, mediaQuery } from '../../tokens';

export const Wrapper = styled.div`
    align-items: center;
    display: flex;
    height: calc(100vh - ${MOBILE_HEADER_HEIGHT});
    justify-content: center;

    @media ${mediaQuery.tablet} {
        height: calc(100vh - ${HEADER_HEIGHT});
    }
`;
