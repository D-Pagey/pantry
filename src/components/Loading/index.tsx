import React from 'react';
import PropagateLoader from 'react-spinners/PropagateLoader';
import { colours } from '../../tokens';
import * as S from './styles';

type LoadingTypes = {
    isLoading: boolean;
};

export const Loading = ({ isLoading }: LoadingTypes): JSX.Element => (
    <S.Wrapper data-testid="loading">
        <PropagateLoader color={colours.darkGreen100} loading={isLoading} />
    </S.Wrapper>
);
