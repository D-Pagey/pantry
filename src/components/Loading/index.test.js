import React from 'react';
import Loading from '.';

describe('Loading component', () => {
    it('should render', () => {
        const { container } = render(<Loading />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
