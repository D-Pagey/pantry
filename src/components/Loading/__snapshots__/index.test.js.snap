// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc(100vh - 100px);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
  data-testid="loading"
>
  <div
    class="css-9ebb9a"
  >
    <div
      class="css-27lg9n"
    />
    <div
      class="css-sr5mcw"
    />
    <div
      class="css-1u9enze"
    />
    <div
      class="css-1r22dar"
    />
    <div
      class="css-10wmy8t"
    />
    <div
      class="css-15wz10t"
    />
  </div>
</div>
`;
